src/patch_parser.py

```diff
@@ -390,28 +428,39 @@ def _parse_patch_file_unchanged_chunk(
                                   side='',
                                   offset=(gerrit_new_line - lines.line_number())))
 
+
 def _parse_patch_file_added_chunk(
         lines: InputSource,
         gerrit_orig_line: int,
         gerrit_new_line: int) -> Tuple[int, int, PatchFileChunkLineMap]:
     in_start = lines.line_number()
     logging.info('First char - 1: %c', lines[0][0])
     while lines[0] and lines[0][0] == '+':
+        lines.set_previous_item(lines[0])
+        previous = lines.get_previous_item()
         lines.consume()
         gerrit_orig_line += 1
         gerrit_new_line += 1
+        if previous[0] == '-': # maybe add check to see if it is actually a modified one though I think this situation only applies when it actually is modified
```

Comment:
I think this might need to be generalize to handle chunks like ++-- and so on.

Perhaps we can pass in the previous chunk into this function.
The loop that calls this builds up a `chunks` list so we should be able to pass in an `Optional[...]` there.

We can refactor these inputs (and return values) into a struct so we're not passing around a bigger and bigger tuple (we maybe should have done that to start with, tbh).

Reply:
+1